facia-tool/app/tools/FaciaToolArchive.scala

```diff
@@ -0,0 +1,54 @@
+package tools
+
+import com.amazonaws.regions.{Regions, Region}
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsyncClient
+import com.amazonaws.services.dynamodbv2.model._
+import common.{ExecutionContexts, Logging}
+import conf.{Switches, Configuration}
+import org.joda.time.{DateTimeZone, DateTime}
+import play.api.libs.json._
+import scala.collection.JavaConverters._
+import awswrappers.dynamodb._
+
+import scala.util.{Failure, Success}
+
+case class ArchiveRequest(email: String, updateJson: JsValue, diff: JsValue)
+
+object FaciaToolArchive extends ExecutionContexts with Logging {
+  val TableName = "FaciaToolUpdateHistory"
+  private val dynamoClient: Option[AmazonDynamoDBAsyncClient] =
+    if (Configuration.environment.isProd) {
+      val c = new AmazonDynamoDBAsyncClient()
+      c.setRegion(Region.getRegion(Regions.EU_WEST_1))
+      Option(c)
+    } else None
+
+  def dayKey(date: DateTime) = date.toString("yyyy-MM-dd")
```

Comment:
Just to warn you, I tried doing something similar for the CSS usage thing, and found it was throttling a lot of my requests. I ended up switching out to use the selector as the hash and using the day in the range.

Depends on your use case, but if you're recording history, most of the time I assume people will want to look at history of a particular container (or collection of containers), in which case you could use the container ID as the hash, and then a range key for the datetime. You can easily aggregate multiple containers to get a history for a front. You can also get it to push updates for a table into Kinesis, which you can then do fancier stuff with ...